lib/Auth/Source/scoutnetauth.php

```diff
@@ -109,4 +108,126 @@ protected function login($username, $password)
             throw new SimpleSAML_Error_Error('WRONGUSERPASS');
         }
     }
+
+    /**
+     * Formats a name to be used in an email address.
+     * @param string $name
+     * @return string
+     */
+    private function formatNameForEmail($name)
+    {
+        $name = trim(strtolower($name)); // Convert to lower case and trim spaces.
+        $name = preg_replace('/\s+/', '.', $name); // Replace empty space with one dot each.
+        $name = preg_replace('/[.][\-]/', '-', $name); // Replace .- with -
```

Comment:
why use character-groups with just one char in it?
'/[.][\-]/' = '/\.\-/'
And why use regexp when there are just static strings?
`strtr($name, ['.-' => '-', '-.' => '-']);`

Reply:
That's what [Google-Scoutnet-sync](https://github.com/scouternasetjanster/Google-Scoutnet-synk/blob/68e9bd9669eb7abc3c526a761a003b0105a796b6/Anvandare.gs#L258) does.
I'll change it with the bug you mentioned.